src/gadget/ecc.rs

```diff
@@ -0,0 +1,67 @@
+//! Gadgets and chips for elliptic curve operations.
+
+use std::fmt;
+
+use crate::{
+    arithmetic::Curve,
+    circuit::{Chip, Layouter},
+    plonk::Error,
+};
+
+/// The set of circuit instructions required to use the ECC gadgets.
+pub trait EccInstructions<C: Curve>: Chip<Field = C::Base> {
```

Comment:
I think `C` here needs to be a type parameter rather than an associated type, because it's _possible_ (though unlikely) that a chip may want to implement `EccInstructions` for multiple curves built over the same base field.

Reply:
Actually that's not so unlikely, e.g. it's needed for simplified SWU (to implement the isogenous curve).